magda-csw-connector/src/Csw.ts

```diff
@@ -81,14 +87,20 @@ export default class Csw implements ConnectorSource {
 
             for (let i = 0; i < records.length; ++i) {
                 const recordXml = records.item(i);
-                result.push(this.xmlRecordToJsonRecord(recordXml));
+                result.push(new DatasetContainer(this.xmlRecordToJsonRecord(recordXml),
+                                        moment(
+                                            parseInt(
+                                                pageXml.documentElement.getElementsByTagName("retrievedBy")[0].nodeValue)
```

Comment:
Shouldn't it be `retrievedAt` instead of `retrievedBy`?  Were you able to test this?